components/TextArea.tsx

```diff
@@ -35,67 +40,49 @@ const TextArea = () => {
             spellCheck="false"
             data-enable-grammarly="false"
          /> */}
-         {wordList?.map((word, index) => {
-            const isActive =
-               activeWord === word && typedHistory.length === index;
+         {wordList?.map((word, wordIndex) => {
+            const isActive = activeWord === word && typedHistory.length === wordIndex;
+            const typedWord = isActive ? userInput : typedHistory[wordIndex] || '';
+            
             return (
                <div
-                  key={word + index}
+                  key={word + wordIndex}
                   className="relative mt-0 mx-[7px] mb-1"
                   ref={isActive ? activeWordRef : null}
                >
-                  <div className="startView">
-                     {isActive ? (
-                        <span
-                           ref={caretRef}
-                           id="caret"
-                           className="animate-blink rounded-sm flex items-start w-[.08em] h-7 top-1 bg-cursor justify-start text-cursor  absolute"
-                           style={{
-                              left: userInput.length * 12.3833,
-                           }}
-                        />
-                     ) : null}
-                  </div>
+                  {isActive && (
+                     <span
+                        ref={caretRef}
+                        id="caret"
+                        className="animate-blink rounded-sm flex items-start w-[.08em] h-7 top-1 bg-cursor justify-start text-cursor absolute"
+                        style={{
+                           left: typedWord.length * 12.3833,
+                        }}
+                     />
+                  )}
                   {word.split('').map((char, charIndex) => {
-                     const isCorrectlyTyped =
-                        isActive && char === userInput[charIndex];
+                     const typedChar = typedWord[charIndex];
+                     const isCorrect = typedChar !== undefined ? char === typedChar : null;
+                     
                      return (
                         <span
                            key={char + charIndex}
-                           className={isCorrectlyTyped ? 'text-yellow-500' : ''}
+                           style={getCharStyle(isCorrect)}
```

Comment:
use classnames please

`className = {isCorrect ? 'text-correct_char' : 'text-wrong_char'}`
something like this should work.

Try this let's see.. but just use className over here.